eng/jobs/finalize-publish.yml

```diff
@@ -0,0 +1,94 @@
+parameters:
+  Jobs: []
+jobs:
+- ${{ each job in parameters.Jobs }}: 
+  - job: ${{ job.displayName }}
+    displayName: ${{ job.displayName }}
+    # Run only if all build legs succeeded
+    condition: and(succeeded('Linux_x64_glibc'),
+                  succeeded('Linux_Arm64'),
+                  succeeded('Linux_Arm'),
+                  succeeded('Linux_x64_Alpine36'),
+                  succeeded('Linux_x64_Rhel6'),
+                  succeeded('Windows_x64'),
+                  succeeded('Windows_x86'),
+                  succeeded('Windows_Arm'),
+                  succeeded('Windows_Arm64'),
+                  notIn(variables['Build.Reason'], 'IndividualCI', 'BatchedCI', 'PullRequest'))
+    # Run after all dependent legs are executed
+    dependsOn: ${{ job.dependsOn }}
+    pool: ${{ job.pool }}
+    variables: 
+      ${{ insert }}: ${{ job.variables }}
+    steps:
+
+    # Initialize tooling
+    - script: build.cmd
+        -- /t:BuildTraversalBuildDependencies
+      displayName: Initialize tooling
+      condition: succeeded()
+
+    - task: MicroBuildSigningPlugin@2
+      displayName: Install MicroBuild plugin for Signing
+      inputs:
+        signType: $(SignType)
+        zipSources: false
+        feedSource: https://dnceng.pkgs.visualstudio.com/_packaging/MicroBuildToolset/nuget/v3/index.json
+      continueOnError: false
+      condition: and(succeeded(), in(variables['SignType'], 'real', 'test'))
+
+    - task: MSBuild@1
+      displayName: Publish (no PublishType)
+      inputs: 
+        solution: $(Build.SourcesDirectory)\publish\publish.proj
+        platform: $(_TargetArchitecture)
+        configuration: $(_BuildConfiguration)
+        msbuildVersion: 15.0
+        msbuildArchitecture: $(_TargetArchitecture)
+        msbuildArguments: '/p:Finalize=true 
+        /p:OfficialPublish=true 
+        /p:PublishType=$(PublishType) 
```

Comment:
Here's an example of where there could be weird behavior because if you removed `/p:PublishType=$(PublishType)` your build would exhibit the same behavior because PublishType would still get set.

Reply:
Will take care of this.